Establish a mechanism that allows for other revocation means in the Referenced Token (#90)
vcstuff/draft-ietf-oauth-status-list: draft-ietf-oauth-status-list.md

1. The claim value MUST be a valid JSON object.

> 2. The claim value object MUST contain a member called "status_list" that signals that status checks for this token can be done using the status list mechanism defined in this document. For the "status_list" object, the following rules apply:

Reviewer: If we make this a MUST then won't other future mechanisms only be additional rather than alternatives?

Reviewer 2: this is only a MUST for the mechanism used in the specification. If Giuseppe wants to implement an OCSP-like mechanism than it would look mlike:
```json
"status" : {
  "new-mechanism-name" : {
    "whatever-parameter" : "whatever-value"
  }
}  
```

Reviewer 2: I think we may need some text to the introduction that explains this in more detail. However its more better to merge #89 first and then start working from there

The following content applies to the JWT Claims Set:

* `iss`: REQUIRED. The `iss` (issuer) claim MUST specify a unique string identifier for the entity that issued the Referenced Token. In the absence of an application profile specifying otherwise, compliant applications MUST compare issuer values using the Simple String Comparison method defined in Section 6.2.1 of {{RFC3986}}. The value MUST be equal to that of the `iss` claim contained within the referenced Status List Token.
* `status`: REQUIRED. The `status` (status) claim MUST specify a JSON Object that contains a reference to a status mechanism.
  * `status_list`: REQUIRED when the status list mechanism defined in this specification is used. It contains a reference to a Status List or Status List Token. The object contains exactly two claims:

>     * `idx`: REQUIRED. The `idx` (index) claim MUST specify an Integer that represents the index to check for status information in the Status List for the current Referenced Token. The value of `idx` MUST be a non-negative number, containing a value of zero or greater.

Reviewer 2: Maybe swap the order of Uri and idx?

Author: Current ordering is alphabetical which matches the examples and what afaik most json libraries will generate - that was my main intention for this ordering

# Referenced Token {#referenced-token}

## Status Claim {#status-claim}

> By including a "status" claim in a JWT, the issuer of the JWT declares that the credential is referencing a mechanism to retrieve status information about this credential. The claim contains members used to reference to a status list as defined in this specification. Other members of the "status" object may be defined because status list means to check the status of a credential. This is analogous to "cnf" claim in Section 3.1 of {{RFC7800}} in which different authenticity confirmation methods can be included.

Reviewer 2: it seems this is in the wrong chapter?

> By including a "status" claim in a JWT, the issuer of the JWT declares that the credential is referencing a mechanism to retrieve status information about this credential. The claim contains members used to reference to a status list as defined in this specification. Other members of the "status" object may be defined because status list means to check the status of a credential. This is analogous to "cnf" claim in Section 3.1 of {{RFC7800}} in which different authenticity confirmation methods can be included.

Reviewer 2: ```suggestion
By including a "status" claim in a Referenced Token, the issuer is referencing a mechanism to retrieve status information about this Referenced Token. The claim contains members used to reference to a status list as defined in this specification. Other members of the "status" object may be defined by other specifications. This is analogous to "cnf" claim in Section 3.1 of {{RFC7800}} in which different authenticity confirmation methods can be included.
```
make it more general

Author: I formulated this for JWT specifically as a starting point. This is similar to the JWT cnf registry. My current understanding is that we need to define 2 registries for "status" - one for JWT and one for CWT.
I guess I should adjust the title to match it - let's discuss in the editors call about this issue imho.
